src/api/allfinished.js

```diff
@@ -265,6 +265,83 @@ const getLeaderHistoryForLevel = async (
   return leaderHistoryWithData;
 };
 
+// checks if the current user can view times for the given
+// level and kuski.
+const authLevel = async (req, LevelIndex, KuskiIndex) => {
+  const level = await levelInfo(+LevelIndex);
+  if (!level || level.Locked) return false;
+
+  const auth = authContext(req);
+
+  if (level.Hidden) {
+    if (KuskiIndex) {
+      // user can see their own times on levels that are hidden.
+      // seems likely that user created the level in this case.
+      return auth.userid && Number(KuskiIndex) === auth.userid;
+    }
+    return false;
+  }
+
+  return true;
+};
+
+const getCrippled = async (type, LevelIndex, limit, KuskiIndex = null) => {
+  const where = {
+    LevelIndex,
+  };
+
+  if (KuskiIndex) {
+    where.KuskiIndex = KuskiIndex;
+  }
+
+  switch (type) {
+    case 'noVolt':
+      where.LeftVolt = 0;
+      where.RightVolt = 0;
+      where.SuperVolt = 0;
+      break;
+    case 'noTurn':
+      where.Turn = 0;
+      break;
+    case 'oneTurn':
+      where.Turn = 1;
```

Comment:
for oneTurn I believe 0 turns also counts

Reply:
Smart thinking